src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -1509,11 +1509,9 @@ int xcldev::device::getXclbinuuid(uuid_t &uuid) {
     return 0;
 }
 
-bool xcldev::device::isHostMem(const char *m_tag)
+bool xcldev::device::isHostMem(const unsigned char *m_tag)
 {
-    std::string str(m_tag);
-
-    return (!str.compare(0, 4, "HOST"));
+    return (!strncmp((const char *)m_tag, "HOST", 4));
```

Comment:
please use `reinterpret_cast<const char*>`

Reply:
Why is this a class member function?  It is not part the device interface.  It would be better to have this be a namespace scoped free function.
```
namespace xcldev {
inline bool
isHostMem(const char* tag)
{
  return strcmp(tag,"HOST")==0;
}

inline bool
isHostMem(const unsigned char* tag)
{
  return isHostMem(reinterpret_cast<const char*>(tag));
}
}
```